build/Build.Steps.cs

```diff
@@ -66,15 +66,43 @@ partial class Build
         .Unlisted()
         .Executes(() => ControlFlow.ExecuteWithRetry(() =>
         {
+            var projectsToRestore = Solution.GetCrossPlatformManagedProjects();
+
             if (IsWin)
             {
-                DotNetRestore(s => s
-                    .SetProjectFile(Solution)
-                    .SetVerbosity(DotNetVerbosity.Normal)
-                    .SetProperty("configuration", BuildConfiguration.ToString())
-                    .When(!string.IsNullOrEmpty(NugetPackageDirectory), o =>
-                        o.SetPackageDirectory(NugetPackageDirectory)));
+                projectsToRestore = projectsToRestore.Concat(Solution.GetWindowsOnlyTestApplications());
+            }
 
+            foreach (var project in projectsToRestore)
+            {
+                if (TestPackageVersions.Versions.TryGetValue(project.Name, out var testedPackageVersions))
```

Comment:
As we add this to more projects we may want it disabled by default and run the test against a single version. Anyway, I'm unsure of the current impact.

Reply:
It is good candidate for improvement.
I have a branch with additional changes related to cover all other our library-depended test application by this tool.
The results 
![image](https://user-images.githubusercontent.com/5972917/220537473-988247de-fa12-4a38-8c9f-a303a7e035d7.png)

And the results for the latest main execution:
![image](https://user-images.githubusercontent.com/5972917/220537539-5958ce97-1ce5-493d-9a1e-9b4772e970de.png)

The problem is probably ASP.NET core build/.NET Framework, not the few more test cases.

Reply:
Windows is already pretty bad independent of the current change.